lib/ramble/ramble/software_environments.py

```diff
@@ -497,28 +569,16 @@ def _define_templates(self):
 
         if namespace.packages in self._software_dict:
             for pkg_template, pkg_info in self._software_dict[namespace.packages].items():
-                spec = pkg_info["pkg_spec"] if "pkg_spec" in pkg_info else pkg_info["spec"]
-                compiler = (
-                    pkg_info["compiler"]
-                    if "compiler" in pkg_info and pkg_info["compiler"]
-                    else None
-                )
-                compiler_spec = (
-                    pkg_info["compiler_spec"]
-                    if "compiler_spec" in pkg_info and pkg_info["compiler_spec"]
-                    else None
-                )
-                new_pkg = TemplatePackage(
-                    pkg_template, spec, compiler=compiler, compiler_spec=compiler_spec
-                )
+                new_pkg = TemplatePackage(pkg_template, pkg_info)
                 self._package_templates[pkg_template] = new_pkg
 
         if namespace.environments in self._software_dict:
             for env_template, env_info in self._software_dict[namespace.environments].items():
                 if namespace.external_env in env_info and env_info[namespace.external_env]:
                     # External environments are considered rendered
                     new_env = ExternalEnvironment(env_template, env_info[namespace.external_env])
-                    self._rendered_environments[env_template] = new_env
+                    # TODO: is external env a spack-only concept?
+                    self._rendered_environments["spack"][env_template] = new_env
```

Comment:
I am not sure yet how to handle `ExternalEnvironment` here.

Reply:
I think other package managers might have a similar concept, but I don't exactly know how it'll work.

For example, an external spack environment is somewhat similar to a python venv that is manually set up.

Reply:
For now I hard-code "spack" for `ExternalEnvironment`. Definitely open for suggestions and I can update that in this PR, or in a separate one in the future.